include/ps/internal/message.h

```diff
@@ -90,6 +107,8 @@ struct Node {
   std::string hostname;
   /** \brief the port this node is binding */
   int port;
+  /** \brief whether this node is created by failover */
+  std::atomic<bool> is_recovery;
```

Comment:
but if we don't change the value during reading, then it is fine.

On Tue, Jul 12, 2016 at 7:06 PM, Yizhi Liu <notifications@github.com> wrote:

> In include/ps/internal/message.h
> <https://github.com/dmlc/ps-lite/pull/59#discussion_r70555034>:
>
> > @@ -90,6 +107,8 @@ struct Node {
> >    std::string hostname;
> >    /** \brief the port this node is binding */
> >    int port;
> > +  /** \brief whether this node is created by failover */
> > +  std::atomic<bool> is_recovery;
>
> is_recovery is read at Postoffice::is_recovery() (postoffice.h line 137),
> which may be from another thread.
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/dmlc/ps-lite/pull/59/files/5b6075bcf3918df64686c6614302a88410c23d32#r70555034>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAZv4bQniSO0n9BgcRYiLy8b1qXeMteMks5qVEgfgaJpZM4JI1yk>
> .
>